File: pa5/DictionaryTest.c
//Steven Cruz
//sicruz
//CS12B
//PA55


//-----------------------------------------------------------------------------
// DictionaryTest.c
// Test client for the Dictionary hash
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char *argv[])
{
    Dictionary A = newDictionary();
    char *k;
    char *v;
    char *word1[] = {"ichi", "ni", "san", "shi", "go", "roku", "shichi", "hachi", "ku"};
    char *word2[] = {"uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"};
    int i;

    printf("%s\n", (isEmpty(A) ? "true" : "false"));

    for (i = 0; i < 9; i++)
    {
        insert(A, word1[i], word2[i]);
    }

    printDictionary(stdout, A);

    for (i = 0; i < 9; i++)
    {
        k = word1[i];
        v = lookup(A, k);
        printf("key=\"%s\" %s\"%s\"\n", k, (v == NULL ? "not found " : "value="), v);
    }

    delete(A, "ichi");
    delete(A, "roku");
    delete(A, "go");
    delete(A, "hachi");

    printf("%s\n", (isEmpty(A) ? "true" : "false"));
    printf("%d\n", size(A));

    printDictionary(stdout, A);

    for (i = 0; i < 7; i++)
    {
        k = word1[i];
        v = lookup(A, k);
        printf("key=\"%s\" %s\"%s\"\n", k, (v == NULL ? "not found " : "value="), v);
    }

    // delete(A, "go");  // error: key not found

    printf("%s\n", (isEmpty(A) ? "true" : "false"));
    printf("%d\n", size(A));
    makeEmpty(A);
    printf("%s\n", (isEmpty(A) ? "true" : "false"));

    freeDictionary(&A);

    return (EXIT_SUCCESS);
}

File: labs/lab7/DictionaryTest.java
public class DictionaryTest{

public static void main(String[] args){
      String v;
      Dictionary A = new Dictionary();
      A.insert("1","a");
      A.insert("2","b");
      System.out.println(A.isEmpty());
      System.out.println(A.size());
      System.out.println(A);

      v = A.lookup("1");
      System.out.println("key=1 "+(v==null?"not found":("value="+v)));
      v = A.lookup("3");
      System.out.println("key=3 "+(v==null?"not found":("value="+v)));
      v = A.lookup("7");
      System.out.println("key=7 "+(v==null?"not found":("value="+v)));
      v = A.lookup("8");
      System.out.println("key=8 "+(v==null?"not found":("value="+v)));
      System.out.println();

      A.delete("1");
      System.out.println(A);

      System.out.println(A.isEmpty());
      System.out.println(A.size());
      A.makeEmpty();
      System.out.println(A.isEmpty());
      System.out.println(A);
  }
}